### ISSUE TITLE
Deprecate --profile

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#16799]
The `--profile` feature has not been touched for a long time, and its test coverage is low. Its functionality has been covered and superseded by build scans and Gradle Profiler. It's time to remove it.

Here's a potential roadmap to do so:

1. in Gradle 7.1 we emit a deprecation warning if `--profile` is specified, but still produce `profile.html`,
2. in Gradle 7.x (perhaps in 7.3) we stop producing `profile.html`, but don't fail the build when `--profile` is specified,
3. in Gradle 8.0 we remove the switch completely, causing a build to fail on startup if `--profile` is specified.

---
cc: @gradle/execution

**Comment by oesolutions** [comment]
Any thoughts on making Gradle Profiler an easier/builtin thing that replaces `--profile`? I don't use scans for corporate projects and `--profile` is handy when something feels slow and I want a quick overview.

**Comment by lptr** [comment]
How could we make Gradle Profiler easier to use? You can install it via SDKMAN for example. Does that help?

**Comment by oesolutions** [comment]
You're right, it is already pretty easy to use. My initial thoughts were just making it so you didn't need to install the separate tool and could just invoke the profiling from the gradle command line. It's likely not worth it though.

**Comment by donat** [comment]
The deprecation seems simple enough. I've assigned this issue to the 7.2 release, let's see if it makes the cut.

**Comment by somayaj** [comment]
Hi @ljacomet are we deprecating the build-profile section or just deprecating the CLI option?
